docs/csharp/roslyn-sdk/source-generators-overview.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Source Generators
+description: Source Generators is a C# compiler feature that lets C# developers inspect user code as it is being compiled and generates new C# source files on the fly that are added to the user's compilation.
+ms.date: 06/11/2021
+ms.topic: tutorial
+ms.custom: mvc, vs-dotnet, source-generators
+---
+# Source Generators
+
+This article provides an overview of Source Generators that ships as part of the .NET Compiler Platform ("Roslyn") SDK. Source Generators is a C# compiler feature that lets C# developers inspect user code as it is being compiled and generates new C# source files on the fly that are added to the user's compilation.
+
+## What is a Source Generator?
+
+A Source Generator is a piece of code that runs during compilation and can inspect your program to produce additional source files that are compiled together with the rest of your code.
+
+A Source Generator is a new kind of component that C# developers can write that lets you do two major things:
+
+1. Retrieve a Compilation object that represents all user code that is being compiled. This object can be inspected, and you can write code that works with the syntax and semantic models for the code being compiled, just like with analyzers today.
+
+2. Generate C# source files that can be added to a Compilation object during the course of compilation. In other words, you can provide additional source code as input to a compilation while the code is being compiled.
+
+When combined, these two things are what make Source Generators so useful. You can inspect user code with all of the rich metadata that the compiler builds up during compilation, then emit C# code back into the same compilation that is based on the data you’ve analyzed! If you’re familiar with Roslyn Analyzers, you can think of Source Generators as analyzers that can emit C# source code.
+
+Source generators run as a phase of compilation visualized below:
+
+![Source Generator Visualization](media/source-generators/source-generator-visualization.png)
```

Comment:
Docs uses a better image synatx, you can easily generate these with the [docs authoring pack extension](https://docs.microsoft.com/en-us/contribute/how-to-write-docs-auth-pack) for VSCode.

```suggestion
:::image type="content" source="media/source-generators/source-generator-visualization.png" alt-text="Graphic describing the different parts of source generation":::
```

Also, you shouldn't reference images in different article folders, create a folder for this article in the `media` directory and copy the images you need.